### ISSUE TITLE
[Feature]: Vector Queries

### ISSUE THREAD

**Issue by evantkchong** [milvus-io/milvus-sdk-go#319]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

The documentation in the [Milvus User Guide](https://milvus.io/docs/v2.1.x/query.md) indicates that the Milvus GO SDK supports vector queries like so:

```go
queryResult, err := milvusClient.Query(
	context.Background(),                                   // ctx
	"book",                                                 // CollectionName
	"",                                                     // PartitionName
	entity.NewColumnInt64("book_id", []int64{2,4,6,8}),     // expr
	[]string{"book_id", "book_intro"}                       // OutputFields
)
if err != nil {
	log.Fatal("fail to query collection:", err.Error())
}

```

However, the [Milvus Client interface](https://github.com/milvus-io/milvus-sdk-go/blob/master/client/client.go#L34-L147) here has no such method with the [closest method being `QueryByPks`](https://github.com/milvus-io/milvus-sdk-go/blob/a4f6dcc718701d2e7bf86c2948e353af86f78364/client/client_grpc_data.go#L301-L363). What gives? I'd think this would be one of the core features that the SDK supports :(

### Describe the solution you'd like.

Support for plain-ol vector queries, similar to what the documentation suggests

### Describe an alternate solution.

Remove the example from the documentation and state that only Query by IDs is supported in golang

**Comment by congqixia** [comment]
Hi @evantkchong
Thanks for this issue. Currently Milvus 2.1.X does not support query expression other than `pk in [values...]` term. So go sdk only provide `query by pk` API. Query by expression API will be fully released after the Milvus support it.

**Comment by evantkchong** [comment]
Hey @congqixia, thanks for responding! 

Is it a matter of Milvus 2.1.X not officially supporting the feature yet? I was using Milvus CLI to inspect Milvus 2.1.4 and was able to use the query expressions I needed. I feel that it's an important feature and that even if you add it provisionally and declare to users "use at your own risk" or "currently, not all query expressions may be supported", it would still be extremely helpful for users looking to adopt Milvus.

If you're agreeable, I'd be willing to contribute a PR to add it in

**Comment by congqixia** [comment]
@evantkchong looks like you're using the `scalar field filtering` feature? 
`Expr` could be like `key > 1000 or key <= 2000` when executing `search` (aka KNN) operation, which means you have to provide a vector to compare with and have a `top K` parameter specified.
If that's the case, maybe `Search` method will satisfy your requiement?

**Comment by evantkchong** [comment]
Yes, I need to [retrieve vectors via scalar filtering](https://milvus.io/docs/v2.1.x/query.md).

I am aware that the `Search` method allows specifying a boolean expression, but in my use case, I do not have a vector to search with, hence my desire for a bonafide `Query` method like the [Documentation](https://milvus.io/docs/v2.1.x/query.md?go) mentions 😓 

I suppose I could use `Search` with an arbitrary Vector with the filtering I need but that feels really hacky :/

**Comment by congqixia** [comment]
@evantkchong Just double checked with the core team & pymilvus team. Milvus DO has the ability to `retrieve vector via scalar filtering`. However, there is still a risk to use this feature: the amount of vectors selected by the expression is uncertain. In some cases, query expression may select the whole collection. In this case, Milvus may crash with OOM if the dataset is huge.
It's recommended to use this feature after the whole pagination feature is implemented.

And your contributing PR is ALWAYS welcomed.

**Comment by evantkchong** [comment]
Thanks for replying @congqixia! Perhaps I will try and take a shot at it.
